CRM/Contribute/BAO/ContributionPage.php

```diff
@@ -472,23 +474,22 @@ public static function sendMail($contactID, $values, $isTest = FALSE, $returnMes
    * @param array $fieldTypes
    *
    * @return array
+   *
+   * @throws \CRM_Core_Exception
    */
-  protected static function getProfileNameAndFields($gid, $cid, &$params, $fieldTypes = []) {
+  protected static function getProfileNameAndFields($gid, $cid, $params, $fieldTypes = []) {
     $groupTitle = NULL;
     $values = [];
     if ($gid) {
       if (CRM_Core_BAO_UFGroup::filterUFGroups($gid, $cid)) {
         $fields = CRM_Core_BAO_UFGroup::getFields($gid, FALSE, CRM_Core_Action::VIEW, NULL, NULL, FALSE, NULL, FALSE, NULL, CRM_Core_Permission::CREATE, NULL);
         foreach ($fields as $k => $v) {
           if (!$groupTitle) {
-            $groupTitle = $v["groupTitle"];
+            $groupTitle = $v['groupTitle'];
           }
           // suppress all file fields from display and formatting fields
           if (
-            CRM_Utils_Array::value('data_type', $v, '') == 'File' ||
-            CRM_Utils_Array::value('name', $v, '') == 'image_URL' ||
-            CRM_Utils_Array::value('field_type', $v) == 'Formatting'
-          ) {
+            $v['data_type'] ?? '' === 'File' || $v['name'] ?? '' === 'image_URL' || $v['field_type'] === 'Formatting') {
```

Comment:
When I first looked at this, I thought the precidence of the `??` operators and the `===` operators might be ambiguous and we should add parentheses.

On second look, I see that the array data is coming from `CRM_Core_BAO_UFGroup::getFields` which runs every item through `CRM_Core_BAO_UFGroup::formatUFField` which ensures that `data_type`, `name`, and `field_type` keys are always present. So I don't think the coalesce is needed and this is just another historical artifact of our developers' love-affair with `CRM_Utils_Array::value()` using it places where it isn't needed, "just because".

Reply:
ohh - let's remove!